### ISSUE TITLE
The content of a cell is randomly replaced with old code

### ISSUE THREAD

**Issue by NikTarasov** [microsoft/vscode#130719]
Issue Type: <b>Bug</b>

Not sure how to reproduce this. I was working on the code in one cell for a few hours. Switched to another cell, and after that found out that the code in the previous cell was replaced by some old code wiping out half a day of work

Extension version: 2021.8.1195043623
VS Code version: Code 1.59.0 (Universal) (379476f0e13988d90fab105c5c19e7abc8b1dea8, 2021-08-04T23:14:40.191Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz (8 x 2000)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 3|
|Memory (System)|32.00GB (2.43GB free)|
|Process Argv|--crash-reporter-id 555b6cec-98fa-4a6f-bffe-d043e8245680|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
pythonvspyt700cf:30270857
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscoreces:30322571
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805cf:30301675
pythonvspyt200:30340761
vscextlangct:30333562
binariesv615:30325510
pythonvssor306:30344512
bridge0708:30335490
vstre464:30350172
bridge0723cf:30350415

```

</details>

**Comment by greazer** [comment]
Flagging as important as it represents data loss.

**Comment by NikTarasov** [comment]
A bit more details on how it looked. When I switched back to the cell where the code have been replaced with a much older version (from the same cell), the output of the cell looked correctly, i.e. as if it was generated by the code that has disappeared. CTRL+Z didn't help. I executed the cell again, and the same correct output was given, even though the code in the cell was old/wrong.
Hope it makes sense.

**Comment by NikTarasov** [comment]
Just an update. Noticed more of this weird behaviour. This time the code at the top of each cell was replaced with a line of code from the last cell. Very dangerous behaviour, unsafe to use jupyter in vscode imho.

**Comment by NikTarasov** [comment]
Any updates on the issue? I have zero confidence in this extension obviously, and don't risk doing any work with it, which has been rather disruptive to my workflow.
Would appreciate any comment, thanks.

**Comment by greazer** [comment]
Thanks for checking in Nikita. Without a repeatable repro it's very difficult to determine what might be happening. However, your recent message about code at the top of each cell being replaced with a line of code from the last cell sounds a lot like these:

https://github.com/microsoft/vscode-python/issues/16980
https://github.com/microsoft/vscode/issues/131009

Can you confirm that you formatted the notebook and ended up in this situation? Do you also recall whether the initial bug you opened here may have been related to doing a format on the notebook?

**Comment by NikTarasov** [comment]
Hi Jim. I did in fact use code formatting on the notebook, but I don't remember code disappearing right after I used it. I think that's why I didn't connect the two.

**Comment by greazer** [comment]
Yep, I totally understand. Based on the repros I've seen it's not always obvious that something rotten occurred. Thank you for the clarification. We intend to fix this in short order.
